.github/workflows/pr-dependencies-check.yml

```diff
@@ -0,0 +1,211 @@
+name: PR Dependency Check
+
+on:
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: 'PR number or URL'
+        type: string
+        required: true
+      search_depth:
+        description: 'Dependencies search depth'
+        required: true
+        default: '100'
+        type: choice
+        options: ['100', '200', '500']
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Pytorch repository
+        uses: malfet/checkout@silent-checkout
+        with:
+          repository: pytorch/pytorch
+          fetch-depth: -1
+          submodules: false
+          quiet-checkout: true
+
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+
+      - name: Check PR dependencies
+        run: |
+          SEARCH_DEPTH="${{ github.event.inputs.search_depth }}"
+          
+          PR_NUMBER="${{ github.event.inputs.pr_number }}"
+          # extract PR number from URL
+          PR_NUMBER=$(echo "$PR_NUMBER" | grep -oE '[0-9]+$')
```

Comment:
I'm not a bash master, but above grep wouldn't play nicely with newlines, would it, i.e.:
```bash
$ PR_NUMBER='123\n456'; echo $(echo "$PR_NUMBER" | grep -oE '[0-9]+$') 
123 456
```